server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java

```diff
@@ -125,12 +126,22 @@ public CatalogInfo updateCatalog(String name, UpdateCatalog updateCatalog) {
         }
     }
 
-    public void deleteCatalog(String name) {
+    public void deleteCatalog(String name, boolean force) {
         try (Session session = sessionFactory.openSession()) {
             Transaction tx = session.beginTransaction();
             try {
                 CatalogInfoDAO catalogInfo = getCatalogDAO(session, name);
                 if (catalogInfo != null) {
+                    List<SchemaInfoDAO> schemas = schemaRepository.listSchemas(session, catalogInfo.getId(), Optional.of(100));
```

Comment:
Is 100 the page size or the max result? The logic is wrong if it's the max result

It's inefficient to list all schemas when `force` is not specified. Can we list one schema first to check, and only list others when `force` is specified?

Reply:
Yes, you are correct. 100 currently stands for pageSize. We should leave it empty in order to get all the existing child entities.
I have implemented the similar logic for all entities (level-2, level-3) now :
* First check if a single child entity exists. 
* if not `force` throw exception
* else get a list of all child entities of that type by setting `maxResults` as empty/null and delete all of them.